x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleMappingMetadata.java

```diff
@@ -71,9 +95,24 @@ public static NamedDiff<Metadata.Custom> readDiffFrom(StreamInput streamInput) t
 
     @Override
     public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
+        // role mappings are serialized without their names
         return Iterators.concat(ChunkedToXContentHelper.startArray(TYPE), roleMappings.iterator(), ChunkedToXContentHelper.endArray());
     }
 
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        return ChunkedToXContent.wrapAsToXContent(this).toXContent(builder, params);
```

Comment:
can you help me understand this change ? I am mostly concerned if this will change the format that is persisted to cluster state in any way. I suspect we want to avoid 2 variants of persisted state.

Reply:
Thanks for raising this! It's actually **not** required, and I've removed it.
I needed it only for tests, i.e. because `RoleMappingMetadataXContentSerializationTests` extended `AbstractXContentSerializingTestCase`. But I then found out about `AbstractChunkedSerializingTestCase` so now `RoleMappingMetadata` doesn't need to implement anything new (`ToXContent`).